src/gamma_inc.jl

```diff
@@ -421,39 +421,35 @@ See also: [`gamma_inc(a,x,ind)`](@ref SpecialFunctions.gamma_inc)
 """
 function gamma_inc_taylor(a::Float64, x::Float64, ind::Integer)
     acc = acc0[ind + 1]
-    wk = zeros(30)
-    flag = false
-    apn = a + 1.0
-    t = x/apn
-    wk[1] = t
-    loop = 2
-    for indx = 2:20
-       apn += 1.0
-       t *= x/apn
-       if t <= 1.0e-3
-           loop = indx
-           flag = true
-           break
-       end
-       wk[indx] = t
-    end
-    if !flag
-        loop = 20
+    wk = zeros(20)
+    apn = a
+
+    # compute and store larger terms in wk, to add from small to large
+    t = 1
+    i = 0
+    while i < 20
+        i += 1
+        apn += 1.0
+        t *= x / apn
+        t > 1.0e-3 || break
+        wk[i] = t
     end
+
+    # sum the smaller terms directly
     sm = t
-    tol = 0.5*acc #tolerance
-    while true
-       apn += 1.0
-       t *= x/apn
-       sm += t
-       if t <= tol
-           break
-       end
+    tolerance = 0.5acc
+    while t > tolerance
+        apn += 1.0
+        t *= x / apn
+        sm += t
     end
-    for j = loop-1:-1:1
-       sm += wk[j]
+
+    # sum terms from small to large
+    for v ∈ @view wk[i:-1:1]
+        sm += v
```

Comment:
Using views seems unnecessarily complicated:
```suggestion
    for j in i:(-1):1
        sm += wk[j]
```

And potentially slow as well:
```julia
julia> using BenchmarkTools

julia> function f(x, n)
           s = zero(eltype(x))
           for xi in @view x[n:(-1):1]
               s += xi
           end
           return s
       end
f (generic function with 1 method)

julia> function g(x, n)
           s = zero(eltype(x))
           for i in n:(-1):1
               s += x[i]
           end
           return s
       end
g (generic function with 1 method)

julia> x = rand(10_000_000);

julia> @btime f($x, 10_000);
  28.162 μs (0 allocations: 0 bytes)

julia> @btime g($x, 10_000);
  8.495 μs (0 allocations: 0 bytes)

julia> f(x, 10_000)
5017.769064590679

julia> g(x, 10_000)
5017.769064590679
```